plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt

```diff
@@ -816,7 +817,8 @@ internal class CameraAnimationsPluginImpl : CameraAnimationsPlugin {
    */
   companion object {
     private const val TAG = "Mbgl-CameraManager"
-    internal var animationsPluginWeakRef: WeakReference<CameraAnimationsPluginImpl>? = null
+    internal var pluginRegistry: HashMap<MapTransformDelegate, WeakReference<CameraAnimationsPluginImpl>> =
```

Comment:
Is there a possibility to remove this dependency on companion object fully? Why do we need to have static configuration at all?

Reply:
@tobrun - I can't see any other options to make extension functions work other than have some static registry. We need some kind of link (preferably weak one) to `CameraAnimationsPluginImpl` to declare e.g.:
```
fun MapTransformDelegate.easeTo(
  cameraOptions: CameraOptions,
  animationOptions: MapAnimationOptions? = null
) = CameraAnimationsPluginImpl.pluginRegistry[this]?.get()?.easeTo(cameraOptions, animationOptions)
```
I do not like that approach as well but can't come up with another one, if you have any ideas - would like to hear them!

Reply:
How about add a plugin provider interface into the MapboxMap? so that we can get the plugin instance from the MapboxMap and the extension function should work fine with the MapboxMap instance.

Reply:
@pengdev camera plugin knows actually nothing about `MapboxMap`.
We could enhance `interface MapTransformDelegate` to make it work...
Technically we could add introduce new interface to `sdk-base/src/main/java/com/mapbox/maps/plugin/delegates` and try make it correct way.
Let's track it as separate task.

Reply:
Tracking tailwork here https://github.com/mapbox/mapbox-maps-android/issues/73